clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -765,6 +766,16 @@ public void shouldThrowOnFlushProducerIfProducerIsClosed() {
             fail("Should have thrown as producer is already closed");
         } catch (IllegalStateException e) { }
     }
+    
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldThrowClassCastException() {
+        try (MockProducer<Integer, String> producer = new MockProducer<>(true, new IntegerSerializer(), new StringSerializer());) {
+            ProducerRecord record = new ProducerRecord(topic, "key1", "value1");
+            producer.send(record);
+            fail("Should have thrown ClassCastException because record cannot be casted with serializers");
```

Comment:
instead of `fail` we should use ```
assertThrows(
    ClassCastException.class,
    () -> producer.send(record);
);
```

For this case, we can remove the `catch` block below.

Reply:
Done, I have followed the same approach that other tests in the same class, but assertThrows is also a good alternative

Reply:
Oh, I see. We actually want to pass in an incorrect key-type to get the `ClassCastException`. So you are right, for this case we cannot specify the generics. Keeping the code as-is an to suppress the warning is fine than. Totally missed this.

Reply:
Yes, we used an older version of the test libraries and thus existing code uses a different pattern. We just try to use the newer (and better) pattern in new code and migrate existing code lazily to the new pattern.